src/components/ProgramsTable/ProgramsTable.jsx

```diff
@@ -1,37 +1,103 @@
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import { navigate } from 'gatsby';
+import { StatusAlert } from '@edx/paragon';
+import { faExclamationTriangle } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 
 import Layout from '../Layout/Layout';
 
 class ProgramsTable extends Component {
   constructor(props) {
     super(props);
-    this.state = {
-      programs: [
+    const programData = this.props.programQueryData;
+    this.programData = programData
+      .filter(program => program.node.context.programUUID !== null)
+      .map(program => (
         {
-          uuid: '6eefc008-db50-46f0-8746-667f55533a5d',
-          name: 'Example Program',
-          slug: 'exampleprogram',
-        },
-        {
-          uuid: '6eefc008-db50-46f0-8746-667f55533a5d',
-          name: 'Another Program',
-          slug: 'another-program',
-        },
-      ],
-    };
+          uuid: program.node.context.programUUID,
+          slug: program.node.context.programSlug,
+          name: program.node.context.programName,
+        }
+      ));
   }
+  state = {
+    hasValidPrograms: false,
+  };
 
   componentDidMount() {
-    if (this.state.programs.length === 1) navigate(`${this.state.programs[0].slug}`);
+    // TODO: api call here
+    // this tells us what programs the user is enrolled in so we
+    // can compare it to what programs the deployed application
+    // knows about
+    const enrolledPrograms = [
+      {
+        slug: 'exampleprogram',
+        uuid: '6eefc008-db50-46f0-8746-667f55533a5d',
+      },
+      {
+        uuid: '6eefc008-db50-46f0-8746-667f55533a5d',
+        slug: 'another-program',
+      },
+      {
+        slug: 'missingprogram',
+        uuid: '11111111-bbbb-cccc-dddd-eeeeeeeeeeee',
+      },
+      {
+        slug: 'missingprogram2',
+        uuid: '22222222-bbbb-cccc-dddd-eeeeeeeeeeee',
+      },
+    ];
+
+    this.validateUserPrograms(enrolledPrograms);
+    if (this.state.hasValidProgram === true && this.programData.length < 2) navigate(`${this.programData[0].slug}`);
+  }
+
+  validateUserPrograms(enrolledPrograms) {
+    // list of program uuids that are part of this site
+    const programsList = this.programData.map(program => program.uuid);
+    // list of program uuids that the user is enrolled in
+    const enrolledProgramsList = enrolledPrograms.map(program => program.uuid);
+    // check if the user has program uuid matching the site's program uuids
+    const found = programsList.some(r => enrolledProgramsList.indexOf(r) >= 0);
```

Comment:
nit: You could avoid creating the `found` variable and put its value in the `this.setState` directly.

Reply:
I thought it looked clearer with just setting the state to `found`, but if you really think it should be changed, I can!